ssg/build_yaml.py

```diff
@@ -335,64 +326,141 @@ def __sub__(self, other):
+                    raise ValueError(msg)
+            self.policies[id_] = controls
 
     def resolve_controls(self, controls_manager):
         for policy_id, controls in self.policies.items():
+            to_expand = Control()
+
             if isinstance(controls, list):
-                controls_list = [controls_manager.get_control(policy_id, control_id) for control_id in controls]
+                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
             elif controls == "all":
-                controls_list = controls_manager.get_all_controls(policy_id)
+                to_expand = self._policy_to_items(controls_manager, policy_id)
             else:
                 msg = "Unknown policy content {content} in profile {profile_id}".format(content=controls, profile_id=self.id_)
                 raise ValueError(msg)
-            for control in controls_list:
-                self.selected.extend(control.rules)
-                for varname, value in control.variables.items():
-                    if varname not in self.variables:
-                        self.variables[varname] = value
+
+            self.selected.extend(to_expand.rules)
+            for varname, value in to_expand.variables.items():
+                if varname not in self.variables:
+                    self.variables[varname] = value
+
+    def extend_by(self, extended_profile):
+        self.policies.update(extended_profile.policies)
+        super(ProfileWithSeparatePolicies, self).extend_by(extended_profile)
+
+
+class ProfileWithInlinePolicies(Profile):
+    def __init__(self, * args, ** kwargs):
+        super(ProfileWithInlinePolicies, self).__init__(* args, ** kwargs)
+        self.controls_by_policy = defaultdict(list)
+
+    def apply_selection(self, item):
+        if ":" in item:
+            policy_id, control_id = item.split(":", 1)
+            self.controls_by_policy[policy_id].append(control_id)
+        else:
+            super(ProfileWithInlinePolicies, self).apply_selection(item)
+
+    def resolve_controls(self, controls_manager):
+        for policy_id, controls in self.controls_by_policy.items():
+            if "all" in controls:
+                to_expand = self._policy_to_items(controls_manager, policy_id)
+            else:
+                to_expand = self._controls_to_items(controls_manager, policy_id, controls)
+
+            self.selected.extend(to_expand.rules)
+            for varname, value in to_expand.variables.items():
+                if varname not in self.variables:
+                    self.variables[varname] = value
+
+
+class ResolvableProfile(Profile):
+    def __init__(self, * args, ** kwargs):
+        super(ResolvableProfile, self).__init__(* args, ** kwargs)
+        self.resolved = False
+        self.resolved_selections = set()
+
+    def _control_to_items(self, controls_manager, policy_id, control_id):
+        return controls_manager.get_control(policy_id, control_id)
+
```

Comment:
I think the name is confusing. Also, it doesn't do anything else than calling `get_control`. It's only renaming it.

Reply:
OK, I will discard it.